lucene/core/src/java/org/apache/lucene/util/QueryBuilder.java

```diff
@@ -509,33 +549,40 @@ protected Query analyzeGraphBoolean(String field, TokenStream source, BooleanCla
         end = articulationPoints[i];
       }
       lastState = end;
-      final Query queryPos;
+      final Query positionalQuery;
       if (graph.hasSidePath(start)) {
-        final Iterator<TokenStream> it = graph.getFiniteStrings(start, end);
+        final Iterator<TokenStream> sidePathsIterator = graph.getFiniteStrings(start, end);
         Iterator<Query> queries = new Iterator<Query>() {
           @Override
           public boolean hasNext() {
-            return it.hasNext();
+            return sidePathsIterator.hasNext();
           }
 
           @Override
           public Query next() {
-            TokenStream ts = it.next();
-            return createFieldQuery(ts, BooleanClause.Occur.MUST, field, getAutoGenerateMultiTermSynonymsPhraseQuery(), 0);
+            TokenStream sidePath = sidePathsIterator.next();
+            return createFieldQuery(sidePath, BooleanClause.Occur.MUST, field, getAutoGenerateMultiTermSynonymsPhraseQuery(), 0);
           }
         };
-        queryPos = newGraphSynonymQuery(queries);
+        positionalQuery = newGraphSynonymQuery(queries);
       } else {
-        Term[] terms = graph.getTerms(field, start);
+        List<AttributeSource> attributes = graph.getTerms(start);
```

Comment:
I think I mentioned a List of AttributeSource is weird (I've never seen this) and it's heavyweight.  Why not a TokenStream or TermAndBoost[] ?

Reply:
This is what GraphTokenStreamFiniteStrings returns currently, for multiple tokens at the same position.  Maybe `TermAndBoost[]` would make more sense though.

Reply:
a tentative change is coming in the next commit, I added also few tests to cover that else coding branch